src/scripts/DynamicBackground.ts

```diff
@@ -0,0 +1,134 @@
+class DynamicBackground {
+    constructor() {}
+
+    /*
+    SUN & MOON
+    */
+    static setSunMoonPosition = (date = new Date()) => {
+        const w = window.innerWidth / 1.2;
+
+        // Do the same thing with the height. Responsive = Good times.
+        const h = window.innerHeight / 1;
+
+        // Get the hours and minutes.
+        const hours = date.getHours();
+        const mins = date.getMinutes();
+
+        // Calculate the position of the sun and moon based on the time.
+        const sunRad = (((hours) * 60 + mins) / (24.00 * 62.00)) * Math.PI * 2;
+        const moonRad = (((hours + 12) * 60 + mins) / (24.00 * 60.00)) * Math.PI * 2;
+
+        // Calculate the axis
+        const sunX = (w / 1.8) - (w * Math.sin(sunRad)) / 2;
+        const sunY = (h / 2) + (h * Math.cos(sunRad)) / 2;
+        const moonX = (w / 1.8) - (w * Math.sin(moonRad)) / 2;
+        const moonY = (h / 1.4) + (h * Math.cos(moonRad)) / 2;
+
+
+        // Apply the poisitions based on our previous calculations
+        const sun = document.getElementById('sun');
+        sun.style.top = `${sunY}px`;
+        sun.style.left = `${sunX}px`;
+        const moon = document.getElementById('moon');
+        moon.style.top = `${moonY}px`;
+        moon.style.left = `${moonX}px`;
+    };
+
+    /*
+    SKY & GROUND
+    */
+    static updateBackgrounds = (d = new Date()) => {
+        const hour = d.getHours();
+        const minutes = d.getMinutes();
+        const bgNumber = DynamicBackground.getPicture(hour);
+
+        // Determine starting background images:
+        const bgNumberNext = (bgNumber + 1) % 12;
+
+        // Get opacity (i.e. how far (in percentage) are we in a certain time-block):
+        // Every block is 2 hours, so 1 hour into a block would be 50% (0.50)
+        // If we are in an even hour add 50%
+        let opacity = hour % 2 ? 0 : 0.5;
+        // Every minute would be 1/120th of a block (minutes / 120)
+        opacity += minutes / 120;
+
+        // Set sky image
+        document.getElementById('sky1').classList.value = `sky sky-${bgNumber}`;
+        document.getElementById('sky2').style.opacity = opacity.toString();
+        document.getElementById('sky2').classList.value = `sky sky-${bgNumberNext}`;
+
+        // Set ground image
+        document.getElementById('ground1').classList.value = `ground ground-${bgNumber}`;
+        document.getElementById('ground2').style.opacity = opacity.toString();
+        document.getElementById('ground2').classList.value = `ground ground-${bgNumberNext}`;
+    };
+
+    static updateScene = (date = new Date()) => {
+        try {
+            DynamicBackground.setSunMoonPosition(date);
+            DynamicBackground.updateBackgrounds(date);
+        } catch (e) {}
+    };
+
+    // Determines the images to use based on the hour
+    static getPicture = hour => {
```

Comment:
This is looking an awful lot like `floor((hour - 1) / 2)` with some fudging for hour < 1